edkrepo/commands/checkout_pin_command.py

```diff
@@ -98,23 +98,53 @@ def run_command(self, args, config):
                 sparse_checkout(workspace_path, pin_repo_sources, manifest)
 
     def __get_pin_path(self, args, workspace_path, manifest_repo_path, manifest):
+        pin_path = None
+        if not args.pinfile.endswith('.xml'):
+            pin_name = '{}.xml'.format(args.pinfile)
+        else:
+            pin_name = args.pinfile
+
         if os.path.isabs(args.pinfile) and os.path.isfile(args.pinfile):
-            return os.path.normpath(args.pinfile)
-        elif manifest_repo_path is not None and os.path.isfile(os.path.join(manifest_repo_path, os.path.normpath(manifest.general_config.pin_path), args.pinfile)):
-            return os.path.join(manifest_repo_path, os.path.normpath(manifest.general_config.pin_path), args.pinfile)
-        elif manifest_repo_path is not None and os.path.isfile(os.path.join(manifest_repo_path, args.pinfile)):
-            return os.path.join(manifest_repo_path, args.pinfile)
-        elif os.path.isfile(os.path.join(workspace_path, args.pinfile)):
-            return os.path.join(workspace_path, args.pinfile)
-        elif os.path.isfile(os.path.join(workspace_path, 'repo', args.pinfile)):
-            return os.path.join(workspace_path, 'repo', args.pinfile)
-        elif not os.path.isfile(os.path.join(workspace_path, args.pinfile)) and os.path.dirname(args.pinfile) is None:
-            for dirpath, dirnames, filenames in os.walk(workspace_path):
-                if args.pinfile in filenames:
-                    return os.path.join(dirpath, args.pinfile)
+            pin_path = os.path.normpath(args.pinfile)
+        elif manifest_repo_path is not None:
+            pin_path = self.__find_pin_in_manifest_repo(pin_name, manifest_repo_path, manifest.general_config.pin_path)

```

Comment:
We are losing the `os.path.normpath` for `manifest.general_config.pin_path`. Is this ok?

Reply:
Yes the __find_pin_in_manifest_repo() method will normalize the path before returning it.